pyLucXor/peptide.py

```
@@ -0,0 +1,907 @@
+    def _fragment_ion_mass(self, ion_str, z, addl_mass, ion_type='b'):
+        """
+        Calculate fragment ion mass
+        Receives complete ion string, such as "b4:FLLE" or "y2:sK"
+        """
+        mass = 0.0
+        # Parse amino acid sequence starting after colon
+        start = ion_str.find(":") + 1
+        stop = len(ion_str)
+        seq = ion_str[start:stop]
+        
+        for idx, aa in enumerate(seq):
```

Ruff (0.12.2)

Loop control variable `idx` not used within loop body

Rename unused `idx` to `_idx`

(B007)